src/Raven.Server/Documents/Handlers/ConfigurationHandler.cs

```diff
@@ -5,35 +5,57 @@
 using Sparrow.Json;
 
 namespace Raven.Server.Documents.Handlers
 {
     public class ConfigurationHandler : DatabaseRequestHandler
     {
-        [RavenAction("/databases/*/configuration/studio", "GET", AuthorizationStatus.ValidUser, EndpointType.Read)]
-        public async Task GetStudioConfiguration()
+        internal abstract class AbstractStudioConfigurationHandlerProcessor<TRequestHandler, TOperationContext> : AbstractHandlerProcessor<TRequestHandler, TOperationContext>
+            where TRequestHandler : RequestHandler
+            where TOperationContext : JsonOperationContext
         {
-            var configuration = Database.StudioConfiguration;
+            protected AbstractStudioConfigurationHandlerProcessor(TRequestHandler requestHandler, JsonContextPoolBase<TOperationContext> contextPool) : base(requestHandler, contextPool)
+            { }
 
-            if (configuration == null)
-            {
-                HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
-                return;
-            }
-
-            using (ContextPool.AllocateOperationContext(out JsonOperationContext context))
+            public async Task WriteStudioConfiguration(StudioConfiguration studioConfiguration)
             {
-                var val = configuration.ToJson();
-                var clientConfigurationJson = context.ReadObject(val, Constants.Configuration.StudioId);
+                if (studioConfiguration == null)
+                {
+                    HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
+                    return;
+                }
 
-                await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBodyStream()))
+                using (ContextPool.AllocateOperationContext(out JsonOperationContext context))
                 {
-                    writer.WriteObject(clientConfigurationJson);
+                    var val = studioConfiguration.ToJson();
+                    var clientConfigurationJson = context.ReadObject(val, Constants.Configuration.StudioId);
+
+                    await using (var writer = new AsyncBlittableJsonTextWriter(context, RequestHandler.ResponseBodyStream()))
+                    {
+                        writer.WriteObject(clientConfigurationJson);
+                    }
                 }
             }
         }
 
+        internal class StudioConfigurationHandlerProcessor : AbstractStudioConfigurationHandlerProcessor<DatabaseRequestHandler, DocumentsOperationContext>
+        {
+            public StudioConfigurationHandlerProcessor(DatabaseRequestHandler requestHandler, DocumentsContextPool contextPool) : base(requestHandler, contextPool) { }
+        }
+
+        [RavenAction("/databases/*/configuration/studio", "GET", AuthorizationStatus.ValidUser, EndpointType.Read)]
+        public async Task GetStudioConfiguration()
+        {
+            var configuration = Database.StudioConfiguration;
```

Comment:
Abstract processor should have abstract `GetStudioConfiguration` method and this should be moved there